src/parser/mod.rs

```diff
@@ -10408,13 +10408,75 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let with_fill = if self.parse_keywords(&[Keyword::WITH, Keyword::FILL]) {
+            Some(self.parse_with_fill()?)
+        } else {
+            None
+        };
```

Comment:
We can guard this behavior with the `dialect_of!(self is ClickhouseDialect | GenericDialect)` so that it only applies to clickhouse and the generic dialects